### PR TITLE
Clean up value type vm code

### PR DESCRIPTION
**Pull request by theresa-m**
These are a few miscellaneous clean up items for value types I found when hunting for "Q" uses to clean up.

- Move getClassFileVersion to be available in Java 20+, matching the ri
- Remove stale constant pool class name checks for value types
- Correct formatting for compiler tags

### CODE REVIEW

File: jcl/src/java.base/share/classes/java/lang/Class.java [hunk 1]
```diff
@@ -5802,6 +5788,21 @@ public Set<AccessFlag> accessFlags() {
 		}
 		return AccessFlag.maskToAccessFlags(rawModifiers, location);
 	}
+
+	/**
+	 * Return class file version (minorVersion << 16 + majorVersion) in an int.
+	 *
+	 * @return	the class file version
+	 */
+	int getClassFileVersion() {
```
[the remark this answers is not in the document]
Comment (theresa-m) [reply jcl/src/java.base/share/classes/java/lang/Class.java:5797]: You are right, for 20-23 it is private. In the valhalla repo it is modified to package-private.

[the remark this answers is not in the document]
Comment (theresa-m) [reply jcl/src/java.base/share/classes/java/lang/Class.java:5797]: Fixed.

**Comment by hangshao0** [comment]
Jenkins test sanity,extended alinuxval jdknext

**Comment by hangshao0** [comment]
jenkins test sanity,extended zlinux jdk21

**Comment by hangshao0** [comment]
The Valhalla PR [build](https://openj9-jenkins.osuosl.org/job/Test_openjdkValhalla_j9_sanity.functional_aarch64_linux_valhalla_Personal/7/) failed.

**Comment by hangshao0** [comment]
@theresa-m Are you looking at the failure mentioned in https://github.com/eclipse-openj9/openj9/pull/19732#issuecomment-2179394156 ?

**Comment by theresa-m** [comment]
Yes, I'm working on it.

**Comment by theresa-m** [comment]
The ValueTypeTests should pass now.

**Comment by hangshao0** [comment]
Jenkins test sanity,extended alinuxval jdknext

**Comment by hangshao0** [comment]
Jenkins test sanity,extended zlinux jdk21

**Comment by hangshao0** [comment]
Jenkins test sanity,extended alinuxval jdknext

**Comment by hangshao0** [comment]
Jenkins test sanity,extended amac jdk21

**Review by hangshao0 — APPROVED** [review]
LGTM.

**Pull request merged**